### PR TITLE
Add super blocks and challenge placeholders

### PR DESCRIPTION
**Pull request by BerkeleyTrue**
Change map structure
change full stack to back end
change hike structure
update to new react-motion api
add isComingSoon logic

### CODE REVIEW

**Comment by SaintPeter** [comment]
Well, that's failing CI :smile: 
Does this not have the changes that @QuincyLarson and I made to the Intro to Bonfires and Intro to Ziplines?

File: seed/challenges/00-getting-started/gear-up-for-success.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
         ],
         [
           "http://i.imgur.com/D7Y5luw.jpg",
-          "A graph of the rate of job growth against growth in computer science degree graduates. There are 1.4 million jobs and only 400 thousand people to fill them.",
+          "A graph of the rate of job growth against growth in computer science degree graduates. There are 1.4 million jobs and only 400 million people to fill them.",
```
Comment (ltegman) [inline seed/challenges/00-getting-started/gear-up-for-success.json]: This reverts a recent fix. It should be thousand.

**Comment by bugron** [comment]
Also Spanish translations are missing for some challenges.

File: seed/challenges/00-getting-started/gear-up-for-success.json (hunk 1 above)
Comment (BerkeleyTrue) [inline seed/challenges/00-getting-started/gear-up-for-success.json]: Don't know how this happened. Will fix.

**Comment by BerkeleyTrue** [comment]
@bugron Hopefully that re-adds all the Spanish back in.

File: seed/challenges/02-data-visualization-certification/data-visualization-certificate.json [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        {
+          "properties": ["isHonest", "isDataVisualizationCert"],
+          "apis": ["/certificate/honest", "/certificate/verify/data-visualization"],
+          "stepIndex": [1, 2]
+        }
+      ],
+      "description": [
+        [
+          "http://i.imgur.com/syJxavV.jpg",
+          "An image of our Front End Development Certificate",
```
Comment (bugron) [inline seed/challenges/02-data-visualization-certification/data-visualization-certificate.json]: So is this a Frond End Development certification or Data Visualization? If second one then the entire Spanish translation is wrong too.

**Comment by ltegman** [comment]
This looks really good. The only problem I'm having is that once I complete and submit a challenge I always get kicked back to the map And there's a flash message at the top that says `Once you have completed all of our challenges, you should join our Half Way Club and start getting ready for our nonprofit projects.`. Once that is fixed and the merge conflicts are resolved I'm ready to merge this.

**Comment by BerkeleyTrue** [comment]
@ltegman Thanks for taking the time to QA. 

I was able to fix that issue but now I'm facing another. Will push up commits soon

**Comment by BerkeleyTrue** [comment]
Ready to QA

**Comment by ltegman** [comment]
I was just about to merge when I happened to catch one more problem. When you complete a challenge before one that is "coming soon" it will take you to the unfinished challenge. For example, completing the Mongo waypoint will take you to `Basejump: API Project 1`.

**Comment by BerkeleyTrue** [comment]
@ltegman That is on purpose. Those challenges are accessible during development and are inaccessible in production. To test this `NODE_ENV=production node server/debug-entry.js`.

**Comment by ltegman** [comment]
@BerkeleyTrue In prod it gives a 404 when you complete the challenge, saying it can't load the unfinished challenge. Definitely better than loading the unfinished challenge, but still not a great user experience.

**Comment by BerkeleyTrue** [comment]
@ltegman It's definitely supposed to pretend that challenge doesn't exist. You're saying that it tries to load one of those challenges when one before it is completed?

**Comment by ltegman** [comment]
@BerkeleyTrue Yeah. Here's what I see when completing the mongo challenge in prod mode: 
a_map_to_learn_to_code_and_become_a_software_engineer___free_code_camp

**Comment by BerkeleyTrue** [comment]
@ltegman you are right. I added the log to filter these out to late in the stream for returnNextChallenge. I'll need to update that before this can be merged.

**Comment by BerkeleyTrue** [comment]
Should be fixed now.

**Comment by ltegman** [comment]
Hmm. I'm still getting the 404 on my end when I test with `NODE_ENV=production node server/debug-entry.js`

**Comment by BerkeleyTrue** [comment]
Ok, I think I finally got it. Fingers crossed

**Comment by ltegman** [comment]
@BerkeleyTrue Woot! That did the trick! Merging this now :+1:

**Comment by BerkeleyTrue** [comment]
@ltegman /play secret

**Comment by BerkeleyTrue** [comment]
@ltegman :shipit:

File: common/app/routes/Hikes/components/Question.jsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import React, { PropTypes } from 'react';
-import { Spring } from 'react-motion';
```
Comment (skcam888) [inline common/app/routes/Hikes/components/Question.jsx]: 1000

**Pull request merged**